package.json

```diff
@@ -800,6 +842,31 @@
           ],
           "default": "Diagnostic",
           "description": "Defines the verbosity of output to be used when debugging a test or a block. For Pester 5 and newer the default value Diagnostic will print additional information about discovery, skipped and filtered tests, mocking and more."
+        },
+        "powershell.buttons.RunButtonVisibility": {
+          "type": "boolean",
+          "default": true,
+          "description": "Show the Run button in the editor titlebar."
+        },
+        "powershell.buttons.RunSelectionButtonVisibility": {
+          "type": "boolean",
+          "default": true,
+          "description": "Show the Run Selection button in the editor titlebar."
+        },
+        "powershell.buttons.ClosePanelButtonVisibility": {
+          "type": "boolean",
+          "default": false,
+          "description": "Show the Close panel button in the editor titlebar."
+        },
+        "powershell.buttons.MovePanelLeftButtonVisibility": {
+          "type": "boolean",
+          "default": false,
+          "description": "Show the Move panel left button in the editor titlebar."
+        },
+        "powershell.buttons.MovePanelToBottomButtonVisibility": {
+          "type": "boolean",
+          "default": false,
+          "description": "Show the Move panel to bottom button in the editor titlebar."
```

Comment:
This is really awesome and gives the user a ton of flexibility but I don't think some is going to want just 1 button or just 2 buttons from what appears to be two groups here:

`powershell.buttons.showRunButtons`
`powershell.buttons.showPanelMovementButtons`

Can you change it to these 2 settings and use these everyewhere? I discussed this with @rjmholt and @SydneyhSmith

Reply:
Sure, it still fits my needs so I don't mind limiting the options a bit. I'm just curious why limit it if the work has already been done? I can imagine there's people out there that never use the vertical split view or always use the run selection option that might not care for those buttons.

Reply:
I'm very wary to add configuration because often times people get lost in it. There have been many instances where folks just don't realize that settings already exist or find themselves digging trying to figure out the right configuration for them.

I do want there to be _some_ way to toggle showing these buttons and that's kinda why we want 2 settings instead of 5.

I know the difference is only 3 but we try to have this mindset with all new settings.

>  I can imagine there's people out there that never use the vertical split view or always use the run selection option that might not care for those buttons.

Maybe. But those same people might be just as happy seeing all the buttons... in the off chance that they use the other.

It's in situations like this where I go "let's expose the least amount of stuff that makes sense (in other words 2 settings) and wait for a user to come around to ask for more."

Reply:
I understand and I've fixed it now.

Reply:
Thanks @MartinGC94!